setdate.js

```diff
@@ -0,0 +1,82 @@
+
+var myDate = (function(){
+
+  var __date = null;
+
+  function __checkIfInstantiated(){
+    if (__date) { return __date; } else { return new Date(); };
+  };
+
+  function __milliseconds(){
+    var date = __checkIfInstantiated();
+
+    return date.getTime();
+  };
+
+  function __formatted(){
+    var date = __checkIfInstantiated();
+    var month = getMonthName();
+    var day = date.getDate();
+    var year = date.getFullYear();
+
+    return month + " " + day + ", " + year;
+
+  };
+
+  function setDate(value){
+    value = value || Date.now();
+    __date = new Date(value);
+  };
+
+  function getDate(options){
+    options = options || { format: "milliseconds" };
+    var format = options.format;
+
+    return window["__" + format]();
```

Comment:
You don't want to assume anything about the environment in which your module gets used. In this case, assuming window means I can't run your code in a Node repl and consequently none of the tests I use for evaluating your module work.  Additionally, outside of the variable you're assigning the output of your module to, you want to avoid using the global namespace in this fashion. The module pattern is designed, in part, to encapsulate your code and help prevent usage of the global namespace. Instead, you just want to use functions you've declared inside your module as functions local to your module only (as opposed to attaching them to something like window.) Also, because you have declared your functions inside this module, they aren't actually available on window anyway, were you to run your code in a browser console.